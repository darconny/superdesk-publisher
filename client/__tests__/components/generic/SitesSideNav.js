import React from "react";
import SitesSideNav from "../../../components/generic/SitesSideNav";
import { render, fireEvent } from "@testing-library/react";

const sites = [
  {
    code: "1",
    name: "site1"
  },
  {
    code: "2",
    name: "site2"
  }
];

const selectedSite = {
  code: "2",
  name: "site2"
};

describe("generic/SitesSideNav", () => {
  it("renders correctly open", () => {
    const { container } = render(
      <SitesSideNav
        toggle={jest.fn()}
        setTenant={jest.fn()}
        open={true}
        sites={sites}
        selectedSite={selectedSite}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly closed", () => {
    const { container } = render(
      <SitesSideNav
        toggle={jest.fn()}
        setTenant={jest.fn()}
        open={false}
        sites={sites}
        selectedSite={selectedSite}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("fires toggle", async () => {
    const toggle = jest.fn();

    const { container } = render(
      <SitesSideNav
        toggle={toggle}
        setTenant={jest.fn()}
        open={false}
        sites={sites}
        selectedSite={selectedSite}
      />
    );

    const button = container.querySelector("button.nav-tabs-vertical__link");

    fireEvent.click(button);
    expect(toggle).toHaveBeenCalled();
  });

  it("fires setTenant", async () => {
    const setTenant = jest.fn();

    const { getByText } = render(
      <SitesSideNav
        toggle={jest.fn()}
        setTenant={setTenant}
        open={false}
        sites={sites}
        selectedSite={selectedSite}
      />
    );

    const button = getByText("site1");

    fireEvent.click(button);
    expect(setTenant).toHaveBeenCalled();
    expect(setTenant).toHaveBeenCalledWith(sites[0]);
  });
});
