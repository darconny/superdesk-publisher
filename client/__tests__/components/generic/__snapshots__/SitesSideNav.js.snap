// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic/SitesSideNav renders correctly closed 1`] = `
<div
  class="nav-tabs-vertical"
>
  <ul
    class="nav-tabs-vertical__list"
  >
    <li
      class="nav-tabs-vertical__tab"
    >
      <button
        class="nav-tabs-vertical__link"
      >
        Website
      </button>
    </li>
  </ul>
</div>
`;

exports[`generic/SitesSideNav renders correctly open 1`] = `
<div
  class="nav-tabs-vertical nav-tabs-vertical--closed"
>
  <ul
    class="nav-tabs-vertical__list"
  >
    <li
      class="nav-tabs-vertical__tab"
    >
      <button
        class="nav-tabs-vertical__link"
      >
        Website
      </button>
    </li>
  </ul>
</div>
`;
